cmd/collector/app/sampling/strategystore/interface.go

```diff
@@ -24,6 +24,9 @@ import (
 
 // StrategyStore keeps track of service specific sampling strategies.
 type StrategyStore interface {
+	// Close() from io.Closer  stops the processor from calculating probabilities.
+	io.Closer
```

Comment:
You added this to the interface, but I am not seeing any non-test code that's actually calling it.

Adding Closer to interface it often contentious, some people argue that if you create an object via `NewX() *X`, you already have the ability to call Close on it without adding Close function to the interface that X implements. This doesn't work well when factories are involved since the factory does return an interface, not an actual struct. One other workaround to that is doing a runtime check for `io.Closer` interface and only then calling close - this is why I am asking about prod code calling it.

I'm ok to keep io.Closer in the interface because both real implementations are now closable (static store used to not have close before we added file watcher to it)

Reply:
Yes I think that makes sense.

> This doesn't work well when factories are involved since the factory does return an interface, not an actual struct.

Is there a fundamental reason why factories shouldn't return a struct instead of an interface? (Other than it being a breaking change to make in this instance).

> One other workaround to that is doing a runtime check for io.Closer interface and only then calling close - this is why I am asking about prod code calling it.

Prod code is not calling `Close` - do you have a preference between the current implementation vs the runtime check in tests? I don't feel strongly.

Reply:
> Is there a fundamental reason why factories shouldn't return a struct instead of an interface? 

Yes - polymorphism. The whole point of a factory is to abstract what underlying implementation it creates, which means it always returns an interface.

> Prod code is not calling Close

I actually think our pattern is that the main code only calls factory.Close() and the factory is generally responsible for releasing any resources. E.g. we don't call Close on SpanReader that we obtain from the factory.